### PR TITLE
fix(cli): respect git init.defaultBranch

### PR DESCRIPTION
**Pull request by tinovyatkin**
`projen new` by default initialises a git repository using hard coded default branch name `main`. Some organisations / users may have different defaults for git branch (like `mainline`) and so git since version 2.28 provides global option [init.defaultBranch](https://github.blog/2020-07-27-highlights-from-git-2-28/#introducing-init-defaultbranch). This PR makes `projen new` to respect that setting when present.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tinovyatkin** [comment]
Noticed #2820 but it seems to be stale since July (and with broader changes) and we love this feature to be merged.

File: src/cli/cmds/new.ts [hunk 1]
```diff
@@ -452,10 +458,16 @@ async function initProject(
     );
     logging.debug("system using git version ", gitversion);
     if (gitversion && semver.gte(gitversion, "2.28.0")) {
-      git("init -b main");
+      const defaultGitInitBranch =
+        execOrUndefined("git config init.defaultBranch", {
+          cwd: baseDir,
+        })?.trim() ||
+        args.defaultReleaseBranch ||
```
Comment (mrgrain) [inline src/cli/cmds/new.ts:465]:
```suggestion
        args.defaultReleaseBranch ||
```

Would you be okay to leave this out? I don't really want to mix project settings with cli settings.
The way to get `defaultReleaseBranch` involved is imo to update the default for it to also respect the git setting.

**Review by mrgrain — CHANGES_REQUESTED** [review]
can you add a comment explaining why `main` doesn't need to change here 
(It's because git < 2.28 doesn't have `git config init.defaultBranch`)
```
git("branch -M main");
```

File: src/cli/cmds/new.ts (hunk 1 above)
Comment (tinovyatkin) [reply src/cli/cmds/new.ts:465]: yes, sure

**Review by mrgrain — APPROVED** [review]
Thanks for your contribution! I'm glad we got there in the end. 🚀 
Also thanks to @trilom for the original PR.

**Pull request merged**